test/parallel/test-fs-open.js

```diff
@@ -32,8 +32,20 @@ fs.open(__filename, 'rs', function(err, fd) {
   openFd2 = fd;
 });
 
+try {
+  fs.open('/path/to/file/that/does/not/exist', { bad: 'flags' }, function() {
+    throw new Error('fs.open callback should not have been called');
+  });
+
+  throw new Error('fs.open did not throw despite bad flags');
+} catch (err) {
+  if (!(err instanceof TypeError)) {
+    throw new Error('unexpected error: ' + err);
+  }
+}
```

Comment:
You can still do that like [this](https://github.com/nodejs/node/blob/b5cd2f098691935b6bef6ded1b0de7ef37431f27/test/parallel/test-http-url.parse-only-support-http-https-protocol.js#L48-L56) :)